app/controllers/forms_controller.rb

```diff
@@ -6,6 +6,7 @@ class FormsController < ApplicationController
                                              :thank_you]
   # Since these actions are used to edit forms, maintain the form in session.
   before_action :find_form, only: [:show, :submit, :thank_you, :update]
+  before_action :shibboleth_attributes
```

Comment:
Oh also, if you have a before action in the application controller, you don't need it here too, right?

Or should you have two? One to store the shibboleth attributes (application controller), and one to pull them from the session into an instance variable to be displayed on the page (here)?